src/main/scala/cognite/spark/v1/FlexibleDataModelRelationUtils.scala

```diff
@@ -730,185 +750,139 @@ object FlexibleDataModelRelationUtils {
+  ): Either[CdfSparkException, T] =
     Try {
-      skipNulls(ns).map { n =>
-        val bd = BigDecimal(String.valueOf(n))
-        if (bd.isValidInt) {
-          bd.intValue
-        } else {
-          throw new IllegalArgumentException(s"'${String.valueOf(n)}' is not a valid Int")
-        }
-      }
-    } match {
-      case Success(value) => Right(value)
-      case Failure(e) =>
-        val seqAsStr = ns.map(String.valueOf).mkString(",")
-        Left(new CdfSparkException(s"""Error parsing value for field '$propertyName'.
-                                      |Expecting an Array[Int] but found '[$seqAsStr]' where
-                                      |${e.getMessage}
-                                      |""".stripMargin))
-    }
-
-  private def tryAsFloat(n: Any, propertyName: String): Either[CdfSparkException, Float] = {
-    val nAsStr = String.valueOf(n)
-    val bd = BigDecimal(nAsStr)
-    if (bd.isDecimalFloat) {
-      Right(bd.floatValue)
-    } else {
-      Left(new CdfSparkException(s"""Error parsing value for field '$propertyName'.
-                                    |Expecting a Float but found '$nAsStr'
-                                    |""".stripMargin))
-    }
-  }
+      val asBigDecimal = BigDecimal(String.valueOf(n))
+      safeConvert(asBigDecimal)
+    }.toEither
+      .leftMap(exception =>
+        new CdfSparkException(
+          s"""Error parsing value for field '$propertyName'.
+                                 |Expecting a $expectedType but found '${String.valueOf(n)}'
+                                 |""".stripMargin,
+          exception
+      ))
 
-  private def tryAsFloatSeq(ns: Seq[Any], propertyName: String): Either[CdfSparkException, Seq[Float]] =
+  private def tryConvertNumberSeq[T](
+      ns: Seq[Any],
+      propertyName: String,
+      expectedType: String,
+      safeConvert: (BigDecimal) => T): Either[CdfSparkException, Seq[T]] =
     Try {
       skipNulls(ns).map { n =>
-        val bd = BigDecimal(String.valueOf(n))
-        if (bd.isDecimalFloat) {
-          bd.floatValue
-        } else {
-          throw new IllegalArgumentException(s"'${String.valueOf(n)}' is not a valid Float")
-        }
+        val asBigDecimal = BigDecimal(String.valueOf(n))
+        safeConvert(asBigDecimal)
       }
-    } match {
-      case Success(value) => Right(value)
-      case Failure(e) =>
+    }.toEither
+      .leftMap(exception => {
         val seqAsStr = ns.map(String.valueOf).mkString(",")
-        Left(new CdfSparkException(s"""Error parsing value for field '$propertyName'.
-                                      |Expecting an Array[Float] but found '[$seqAsStr]'
-                                      |${e.getMessage}
-                                      |""".stripMargin))
-    }
+        new CdfSparkException(
+          s"""Error parsing value for field '$propertyName'.
+                                 |Expecting a $expectedType but found '$seqAsStr'
```

Comment:
```suggestion
                                 |Expecting a Array[$expectedType] but found '$seqAsStr'
```
to keep the old format